File: abc086/C.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <boost/range/irange.hpp>

using namespace std;
using namespace boost;

int main() {
    int N;
    cin >> N;
    vector<unsigned int> t(N + 1), x(N + 1), y(N + 1);
    t[0] = x[0] = y[0] = 0;
    for (int i : irange(0, N)) {
        cin >> t[i + 1] >> x[i + 1] >> y[i + 1];
    }
    int t_diff;
    int x_diff;
    int y_diff;
    int l1;
    for (int i : irange(0, N)) {
        t_diff = abs((int) (t[i + 1] - t[i]));
        x_diff = abs((int) (x[i + 1] - x[i]));
        y_diff = abs((int) (y[i + 1] - y[i]));
        l1 = x_diff + y_diff;
        if (t_diff < l1) {
            cout << "No" << endl;
            return 0;
        }
        if (t_diff % 2 != l1 % 2) {
            cout << "No" << endl;
            return 0;
        }
    }
    cout << "Yes" << endl;
    return 0;
}

File: abc086/C.py
import numpy as np


def check(t_diff, x_diff, y_diff):
    l1 = abs(x_diff) + abs(y_diff)
    if l1 > t_diff:
        return False
    elif (t_diff - l1) % 2 == 1:
        return False
    else:
        return True

N = int(input())
t = np.zeros(N+1,)
x = np.zeros(N+1,)
y = np.zeros(N+1,)
for i in range(N+1):
    if i != 0:
        t[i], x[i], y[i] = tuple(map(int, input().split()))

for j in range(N):
    t_diff = t[j+1] - t[j]
    x_diff = x[j+1] - x[j]
    y_diff = y[j+1] - y[j]
    if not check(t_diff, x_diff, y_diff):
        print("No")
        exit()

print("Yes")